pkg/ipcache/types/types.go

```diff
@@ -55,3 +56,19 @@ func NewResourceID(kind ResourceKind, namespace, name string) ResourceID {
 	str.WriteString(name)
 	return ResourceID(str.String())
 }
+
+// IdentityOverride can be used to override the identity of a given prefix.
+// Must be provided together with a set of labels. Any other labels associated
+// with this prefix are ignored while an override is present.
+// This type implements ipcache.IPMetadata
+type IdentityOverride identity.NumericIdentity
```

Comment:
nit: the other types in this package are all used outside the `pkg/ipcache` package, vs.  `pkg/ipcache/types.go` which includes ipcache-internal types. We could probably keep this in the internal-only package given this is only used inside `pkg/ipcache/` for now, plus we already have to update `pkg/ipcache/types.go` to handle this type.